### ISSUE TITLE
Inheriting structs with extra properties

### ISSUE THREAD

**Issue by macnibblet** [go-pg/pg#444]
These are my structs, in some cases i  need to embedded user related data
```go
type Recording struct {
	TableName struct{} `sql:"recordings,alias:recording"`

	Uuid       uuid.UUID `sql:",pk" json:"uuid"`
	ExternalId uint64    `sql:",notnull" json:"externalId"`

	StageName string `json:"stageName"`

	StorageServer string `json:"storageServer"`
	StoragePath   string `json:"storagePath"`

	ViewCount uint `json:"viewCount"`
	LikeCount uint `json:"likeCount"`

	Duration uint16 `json:"duration"`

	UpdatedAt time.Time `json:"updatedAt"`
	CreatedAt time.Time `json:"createdAt"`

	Images []*RecordingImage `json:"images"`

	Performer     *Performer
	PerformerUuid uuid.UUID `sql:",notnull" json:"performerId"`
}

type RecordingWithUserData struct {
	Recording `pg:",override"`

	IsLiked    bool `json:"isLiked"`
	IsFavorite bool `json:"isFavorite"`
}
```

My repository code
```go

func (r *Recording) MatchingWithUserCtx(criteria *infinity.RecordingRepositoryCriteria, user uuid.UUID) ([]infinity.RecordingWithUserData, int, error) {
	var recordings []infinity.RecordingWithUserData

	dataQuery := r.DB.Model(&recordings).Column("recording.*", "Images", "Performer").Order("created_at DESC")
	dataQuery.Apply(r.injectUserCtx(user))
	dataQuery.Apply(r.injectCriteria(criteria, user))

	countQuery := r.DB.Model(new(infinity.Recording))
	countQuery.Apply(r.injectCriteria(criteria, user))

	var err error
	var count int

	wg := sync.WaitGroup{}
	wg.Add(2)

	go func() {
		defer wg.Done()

		if e := dataQuery.Select(); e != nil {
			err = e
		}
	}()

	go func() {
		defer wg.Done()

		var e error
		count, e = countQuery.Count()
		if e != nil {
			err = e
		}
	}()

	wg.Wait()

	return recordings, count, err
}
```

This is the error I'm getting if I'm getting...
```
Starting the servers
panic: reflect: Field index out of range

goroutine 23 [running]:
panic(0x87e6a0, 0xc4205a05f0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
reflect.Value.Field(0x89b6c0, 0xc420592000, 0x199, 0x1, 0x89b6c0, 0xc420592000, 0x199)
	/usr/local/go/src/reflect/value.go:745 +0x1a1
reflect.Value.FieldByIndex(0x91b8c0, 0xc420592000, 0x199, 0xc420562dc0, 0x2, 0x2, 0xc420036800, 0x0, 0x3)
	/usr/local/go/src/reflect/value.go:785 +0xd2
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.(*Field).Value(0xc420565200, 0x91b8c0, 0xc420592000, 0x199, 0x974e6f, 0xc41ffd2e4c, 0x0)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/field.go:43 +0x5f
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.(*Field).AppendValue(0xc420565200, 0x0, 0x0, 0x0, 0x91b8c0, 0xc420592000, 0x199, 0x0, 0xc4200e1ca8, 0xc4205a3640, ...)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/field.go:56 +0x67
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.modelId(0x0, 0x0, 0x0, 0x91b8c0, 0xc420592000, 0x199, 0xc42027a0f8, 0x1, 0x1, 0x928b00, ...)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:140 +0xa6
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.dstValues.func1(0x91b8c0, 0xc420592000, 0x199)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:118 +0xc1
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.visitField(0x91b8c0, 0xc420592000, 0x199, 0xc420563508, 0x0, 0x1, 0xc420187bc8)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:92 +0x108
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.walk(0x91b8c0, 0xc420592000, 0x199, 0xc420563508, 0x0, 0x1, 0xc420187bc8)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:82 +0x18a
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.visitField(0x8d89c0, 0xc420592000, 0x199, 0xc420563500, 0x1, 0x2, 0xc420187bc8)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:90 +0xe2
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.walk(0x875060, 0xc4202ce2c0, 0x197, 0xc420563500, 0x1, 0x2, 0xc420187bc8)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:79 +0x11c
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.dstValues(0x875060, 0xc4202ce2c0, 0x197, 0xc420563500, 0x2, 0x2, 0xc42027a0f8, 0x1, 0x1, 0x0)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/util.go:120 +0x145
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.newManyModel(0xc4201177c0, 0xc420563500)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/model_table_many.go:21 +0x10d
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.(*join).selectMany(0xc4201177c0, 0xb9b520, 0xc4201f3350, 0x921800, 0xc420278370)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/join.go:29 +0xc5
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.(*join).Select(0xc4201177c0, 0xb9b520, 0xc4201f3350, 0x0, 0x536ad0)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/join.go:18 +0xe7
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.selectJoins(0xb9b520, 0xc4201f3350, 0xc4201177c0, 0x2, 0x2, 0xc4205a05e0, 0x0)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/query.go:401 +0x12f
github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm.(*Query).Select(0xc4202531e0, 0x0, 0x0, 0x0, 0x0, 0x45c9a1)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/vendor/gopkg.in/pg.v5/orm/query.go:334 +0x196
github.com/carnagel/infinity-api/pkg/database/postgres.(*Recording).MatchingWithUserCtx.func1(0xc4205635f0, 0xc4202531e0, 0xc4205635d0)
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/pkg/database/postgres/recording.go:157 +0x72
created by github.com/carnagel/infinity-api/pkg/database/postgres.(*Recording).MatchingWithUserCtx
	/home/antoine/Workspace/golang/src/github.com/carnagel/infinity-api/pkg/database/postgres/recording.go:160 +0x356
exit status 2

```

**Comment by vmihailenco** [comment]
Should be fixed in v5.2.1

**Comment by macnibblet** [comment]
Thanks, awesome work as usual! :+1:
